src/lazyloading.js

```diff
@@ -1,53 +1,136 @@
-        el.setAttribute('src', !img ? require('./assets/img/img_skeleton.png') : img);
 
-        const loadImage = () => {
-            el.src = src;
-        };
+/**
+ * Data image default
+ * {Object}
+ * */
+const imageDefault = {
+    src: require('./assets/img/img_skeleton.png'),
+    style: 'height: 170px',
+    class: 's-image',
+};
+/**
+ * Observer options default
+ * {Object}
+ * */
+const observerDefault = {
+    root: null,
+    threshold: 1,
+};
 
-        const callback = (entries, observer) => {
-            entries.forEach((entry) => {
-                if (entry.isIntersecting) {
-                    loadImage();
-                    observer.unobserve(el);
-                }
-            });
-        };
+export default {
+  name: 'CoreLazyLoading',
+  props: {
+    placeholderImage: {
+      type: Object,
+      default: () => {
+        return imageDefault
+      },
+    },
+
+    options: {
+      type: Object,
+      default: () => {
+        return observerDefault
+      },
+    },
+  },
+  data() {
+    return {
+      observer: {},
+      firstStepToLoad: true,
+      image: {
+        class: '',
+        src: '',
+      },
+    }
+  },
+  render() {
+      return this.$slots.default;
+  },
+  mounted() {
+      this.setImageData();
 
-        const lazyImage = () => {
-            const options = {
-                root: null,
-                threshold: 1,
-            };
-            const observer = new IntersectionObserver(callback, options);
-            observer.observe(el);
-        };
+      this.$el.onload = this.onLoadImage;
+      this.$el.onerror = this.onErrorImage;
 
-        el.onload = () => {
-            if (
-                el.getAttribute('src') !==
-                'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAALwAAADGAQMAAABSPnhPAAAAA1BMVEXm5uSlVuPDAAAAG0lEQVR4Ae3BMQEAAADCIPunNsN+YAAAAABAdBNWAAHMZ56dAAAAAElFTkSuQmCC'
-            ) {
-                el.setAttribute('style', '');
-                el.setAttribute('class', className);
-            }
-        };
+      const observer = new IntersectionObserver(
+          this.checkIsIntersecting,
+          this.options
+      );
 
-        el.onerror = () => {
-            el.setAttribute('src', require('./assets/img/photo.png'));
-            el.setAttribute('style', '');
-            el.setAttribute('class', className);
-        };
+      observer.observe(this.$el);
+  },
+  methods: {
+    /**
+     * Start data to component
+     * @returns {null}
+     * */
+    setImageData() {
+          this.image = {
+              src: this.$el.getAttribute('src'),
+              class: this.$el.getAttribute('class'),
+          };
 
-        lazyImage();
+          this.changeElementAttribute(
+              ['class', 'style', 'src'],
+              [
+                  this.placeholderImage.class,
+                  this.placeholderImage.style,
+                  this.placeholderImage.src,
+              ]
+          );
```

Comment:
Looks weird at least. Could be done in a more efficient way. Why can we put the placeholder first and then change it to the original?

Reply:
I thought about it too. I can create two templates